orbmain/src/main/java/com/sun/corba/ee/impl/util/Utility.java

```diff
@@ -302,10 +303,14 @@ static Class loadClassOfType(String className, String remoteCodebase,
                     "expected Type.");
             }
 
-            if (expectedTypeClassLoader != null) {
-                loadedClass = expectedTypeClassLoader.loadClass(className);
-            } else {
-                loadedClass = ORBClassLoader.loadClass(className);
+            try {
+                if (expectedTypeClassLoader != null) {
+                    loadedClass = expectedTypeClassLoader.loadClass(className);
+                } else {
+                    loadedClass = ORBClassLoader.loadClass(className);
+                }
+            } catch (Exception e) {
+                wrapper.classNotFound(className);
```

Comment:
Unclear what happens here. I'm under impression that instance of `RuntimeException` (`org.omg.CORBA.INTERNAL`) is returned from this call, but it is not re-thrown, inspected nor stored anywhere.

Reply:
I am just thinking - what I would expect is that wrapper would distribute the exception to the client/server. Then this solution resolves one issue and might create another.
However Arjan's solution passed all available tests and even fixed one additional issue reported by the user. That's why I would agree to merge this - we did not find any evidence that this fix really breaks anything.

I would just recommend to write some unit tests for the Utility class - at least.

Reply:
I'm not explaining my concerns too well. So here is another try.

What I see:
Now, at line 312 any exception (RTEx like NPEx included) is caught. It's passed to `wrapper.classNotFound` which wraps it into another RTEx (`org.omg.CORBA.INTERNAL`). I observe no side-effect of that. Neither `wrapper.classNotFound` throws, nor after line 313 this very `Utility`. It's swallowed, and at the calling site nothing is known about that. The modified method declares `throws ClassNotFoundException ` and for example in line it's conditionally being re-thrown. But not here.

Some other usages of `wrapper.X` are mixed here - they throw the results or do not.
I'm not 100% familiar with the code, or what it does.

> However Arjan's solution passed all available tests and even fixed one additional issue reported by the user. 

I just do not know what `wrapper.classNotFound(className)` is doing here. Could `return null` be placed here instead?

Reply:
> I just do not know what `wrapper.classNotFound(className)` is doing here.

At this moment I've learnt it's just limited to logging, with ST reduced, in `org.glassfish.pfl.basic.logex.WrapperGenerator#handleFullLogging`. It's the side-effect I was looking for.

Reply:
I think the idea is that it would wrap the logger, and therefore simply log. But it's all quite complicated code.

Reply:
> Could return null be placed here instead?

Absolutely not return null, since that would defeat the entire purpose of the change.

The purpose is.

1. We have loaded a reasonable type.
2. The type is not *perfect*
3. We try to optionally load an even better type
4. There does not seem to be a better type, or the potentially better type cannot be loaded
5. We now lose everything, and don't even try to convert the reasonable type anymore 

So if the optional quest for pure perfection fails, we should return the reasonable type. Only if that reasonable type fails conversion, we should panic and abort.

Reply:
I am finally starting to see that!